packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -264,90 +97,83 @@ const GroupPropertyAction = ({
   const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchInput, setSearchInput] = useState<string>("");
+  const [activeSearchInput, setActiveSearchInput] = useState<string>("");
+  const [searched, setSearched] = useState<boolean>(false);
+  const [activeDragProperty, setActiveDragProperty] = useState<PropertyMetaData>();
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    })
+  );
+  const [showModal, setShowModal] = useState<boolean>(false);
 
-  useEffect(() => {
-    const getContent = async () => {
-      setIsLoading(true);
-      const ruleSet: Ruleset = {
-        id: "element-properties",
-        rules: [
-          {
-            ruleType: RuleTypes.Content,
-            specifications: [
-              {
-                specType: ContentSpecificationTypes.SelectedNodeInstances,
-              },
-            ],
-          }],
-      };
-      const requestOptions: ContentDescriptorRequestOptions<IModelConnection, KeySet, RulesetVariable> = {
-        imodel: iModelConnection,
-        keys: keySet,
-        rulesetOrId: ruleSet,
-        displayType: DefaultContentDisplayTypes.PropertyPane,
-      };
-      const content = await Presentation.presentation.getContentDescriptor(
-        requestOptions
-      );
+  const handleDragStart = (event: DragStartEvent) => {
+    const { active } = event;
+    const activeProperty = selectedProperties.find((p) => active.id === p.key);
+    setActiveDragProperty(activeProperty);
+  };
 
-      // Only primitives and structs for now
-      const properties =
-        content?.fields.filter(
-          (field) =>
-            field.type.valueFormat === PropertyValueFormat.Primitive ||
-            field.type.valueFormat === PropertyValueFormat.Struct
-        ) ?? [];
+  const handleDragEnd = (event: DragEndEvent) => {
+    const { active, over } = event;
 
-      // Map properties to their classes
-      const classToPropertiesMapping = new Map<string, Property[]>();
+    if (over && (active.id !== over.id)) {
+      setSelectedProperties((items) => {
+        const oldIndex = selectedProperties.findIndex((p) => active.id === p.key);
+        const newIndex = selectedProperties.findIndex((p) => over.id === p.key);
 
-      extractProperties(properties, classToPropertiesMapping);
+        return arrayMove(items, oldIndex, newIndex);
+      });
+    }
 
-      const rootClassName = keySet.instanceKeys.keys().next().value;
+    setActiveDragProperty(undefined);
+  };
 
-      // Hardcoded BisCore navigation properties.
-      classToPropertiesMapping.get(rootClassName)?.push({
-        name: "Model.ModeledElement.UserLabel",
-        label: "Model UserLabel",
-        type: "string",
-      });
+  const filteredProperties = useMemo(
+    () =>
+      propertiesMetaData.filter((p) =>
+        [p.label, p.categoryLabel, p.actualECClassName]
```

Comment:
There is a chance that this might produce unexpected results. I think this would be better:
```
[p.label, p.categoryLabel, p.actualECClassName]
  .map((l) => l.toLowerCase())
  .some((l) => l.includes(activeSearchInput.toLowerCase()))
```

Reply:
Oh right, forgot to lowercase.

Reply:
I like this so much better. Thanks :).